app/controllers/reports_controller.rb

```diff
@@ -54,6 +54,18 @@ def selection
 
   def review
     @records = Record.where(id: params[:record_id])
+    @granule_associations = {}
+    @records.each do |record|
+      value = record.associated_granule_value
+      if value.nil? || value == 'Undefined'
```

Comment:
I think there was a question if value could possibly be an empty string. If it is possible, `.nil?` won't catch that, but `.blank?` will.

Reply:
Well it isn't possible for that value to be empty, but regardless I didn't know that, I thought nil? checked for empty as well, so will change it, good to know thanks.